pyiceberg/io/pyarrow.py

```diff
@@ -720,25 +719,17 @@ def primitive(self, primitive: pa.DataType) -> Optional[T]:
 
 
 def _get_field_id(field: pa.Field) -> Optional[int]:
-    for pyarrow_field_id_key in PYARROW_FIELD_ID_KEYS:
-        if field_id_str := field.metadata.get(pyarrow_field_id_key):
-            return int(field_id_str.decode())
-    return None
+    return int(field_id_str.decode()) if (field_id_str := field.metadata.get(PYARROW_FIELD_ID_KEY)) else None
```

Comment:
I realized that we did not check if `field.metadata` is `None` here and may encounter:
```
  File ".../pyiceberg/io/pyarrow.py", line 724, in _get_field_id
    if field_id_str := field.metadata.get(pyarrow_field_id_key):
AttributeError: 'NoneType' object has no attribute 'get'
```
I think this was intended to be fixed in #183. Since that is closed, shall we fix it here?
```suggestion
    return int(field_id_str.decode()) if (field.metadata and (field_id_str := field.metadata.get(PYARROW_FIELD_ID_KEY))) else None
```

Reply:
Again, thanks!